### PR TITLE
Improved Collider Handling

### PR DESCRIPTION
**Pull request by jfc978**
Fixed issues pertaining to boss fireballs and goriya boomerang

### CODE REVIEW

File: CrossPlatformDesktopProject/CrossPlatformDesktopProject/CollisionHandler/CollisionResponse.cs [hunk 1]
```diff
@@ -66,22 +66,35 @@ private void BuildDictionary()
...
+                    commandMap.Add(new Tuple<Type, Type, CollisionSides>(typeof(Boomerang), enemySubject, side), typeof(EnemyTakeDamageCommand));
+                    commandMap.Add(new Tuple<Type, Type, CollisionSides>(typeof(Smoke), enemySubject, side), typeof(EnemyTakeDamageCommand));
                 }
+
+                //Obstacle on player
                 foreach (Type obstacleSubject in obstacleTypes)
                 {
                     commandMap.Add(new Tuple<Type, Type, CollisionSides>(obstacleSubject, playerType, side), typeof(ResetCommand));
                 }
+                commandMap.Add(new Tuple<Type, Type, CollisionSides>(typeof(Bomb), playerType, side), typeof(ResetCommand));
```
Comment (sweeneyde) [inline CrossPlatformDesktopProject/CrossPlatformDesktopProject/CollisionHandler/CollisionResponse.cs:89]: Why is the bomb an automatic death/game reset? And should this be `typeof(Smoke)`?

Comment (sweeneyde) [reply CrossPlatformDesktopProject/CrossPlatformDesktopProject/CollisionHandler/CollisionResponse.cs:89]: Oh nevermind that's not what reset means

**Comment by sweeneyde** [comment]
I think most of this change is fine, but I would be a little worried that it might introduce confusion between "colliders" versus "collections of colliders". The code repetition of `return new List<>{this}` is kind of indicative of that. I feel like keeping the ICollider interface down to just one method might be conceptually simpler. Maybe a separate IColliderCollection interface or something could be useful.

**Comment by jfc978** [comment]
I think we can drop this PR. Roy found a less confusing way to implement it

**Pull request closed without merging**